### PR TITLE
Better word separation

### PR DESCRIPTION
**Pull request by clarkf**
Hello there!

Very neat stuff!  I noticed that squirt wasn't separating words out very intelligently, and wanted to lend a hand.

This pull does two things:
1. It separates words based on all whitespace characters (tab, newline).  Previously, they were split by only spaces.
2. It splits up words with non-whitespace characters, such as punctuation.  Because of the way that `textContent` works, adjacent words that are wrapped in tags (e.g. `<b>This,</b> and this`) will get returned as single words (`['This,and', 'this']`).  This pull adds an extra space after all punctuation, which often helps with words that run-together.

Thanks!

/cc #10, #17

### CODE REVIEW

**Comment by cameron** [comment]
If i had a test suite I'd have no trouble accepting this PR as is (looks obvious enough), but as it stands I want to settle down and poke at a bit before merging. Thank you for taking the time to read and notice! :)

File: bm/squirt.js [hunk 1]
```diff
@@ -229,7 +229,8 @@ sq.host =  window.location.search.match('sq-dev') ?
     return function textToNodes(text) {
       text = "3\n 2\n 1\n " + text.trim('\n').replace(/\s+\n/g,'\n');
       return text
-             .split(' ')
+             .replace(/[\,\.\!\:\;](?![\"\'\)\]\}])/g, "$& ")
+             .split(/[\s]+/g)
```
Comment (bcherry) [inline bm/squirt.js:233]: The character class brackets are unnecessary here

**Comment by bcherry** [comment]
@cameron this isn't really dangerous. the first regex is pretty straightforward, and the replacement is even more so :)

File: bm/squirt.js (hunk 1 above)
Comment (clarkf) [inline bm/squirt.js:233]: @bcherry You mean on line 233 (`/[\s]+/`)?  I believe the class brackets on 232 are necessary.

Comment (bcherry) [inline bm/squirt.js:233]: yeah on 233 they're extraneous

**Comment by clarkf** [comment]
@cameron Totally understood.  Let me know if you'd like any assistance laying the groundwork for a test-suite (I like [mocha](http://visionmedia.github.io/mocha/)+[chai](http://chaijs.com/) personally).

**Comment by walfly** [comment]
This fixes issue #30. Looks good to me, will be excited when its merged.

**Comment by cameron** [comment]
Thanks, @clarkf. I would _adore_ some help with the tests. LMK if you start working on them so I don't do the same :)

**Comment by cameron** [comment]
FYI, Squirt got mentioned on [Business Insider](http://www.businessinsider.com/how-to-start-reading-at-nearly-1000-words-per-minute-today-2014-3) :)

**Comment by aixnr** [comment]
@cameron I'd love to try it. what do I do now? re-drag the bookmarklet from Squirt.IO ?

**Comment by cameron** [comment]
Clear your cache :) The change was to a script that gets injected, not the bookmarklet itself.

**Comment by aixnr** [comment]
@cameron tried with this article http://techcrunch.com/2014/03/11/wikipedia-co-founder-jimmy-wales-accidentally-starts-a-bitcoin-donation-campaign-for-wikipedia/ , hereby I confirmed the fix works. Nice job there @clarkf and @cameron !!

**Comment by clarkf** [comment]
@cameron I'll definitely give it a shot! I'll open a pull shortly to drop commits.

**Pull request merged**